api/lib/domain/usecases/retrieve-last-or-create-certification-course.js

```diff
@@ -12,9 +13,15 @@ module.exports = async function retrieveLastOrCreateCertificationCourse({
   certificationCourseRepository,
   assessmentRepository,
 }) {
-  const { id: sessionId } = await sessionRepository.getByAccessCode(accessCode);
+  const session = await sessionRepository.get(sessionId);
+
+  if (session.accessCode !== accessCode) {
+    throw new NotFoundError();
```

Comment:
Je pense que là ça devrait plutôt être une erreur de type "Forbidden access" que "Not found". Le fait qu'il existe une session avec cet ID n'est plus un secret, l'utilisateur l'a déjà sélectionnée dans un appel précédent.

Reply:
Oui, mais alors c'est un 404 sur session not found, et 403 sur mauvais access code ? J'aimais bien l'idée de mettre le même code de retour dans les deux cas, j'ai pas bien compris la remarque précédente du coup.

Reply:
Oui 404 si tu demandes une session qui n’existe pas, puis 403 si tu essaies de la déverrouiller avec le mauvais code. Pour ma part je ne vois pas pourquoi les deux erreurs seraient considérées de la même façon.